s2rest_play/app/controllers/QueryController.scala

```diff
@@ -1,311 +1,48 @@
 package controllers
 
 
-import com.kakao.s2graph.core.GraphExceptions.BadQueryException
 import com.kakao.s2graph.core._
-import com.kakao.s2graph.core.mysqls._
-import com.kakao.s2graph.core.types.{LabelWithDirection, VertexId}
-import com.kakao.s2graph.core.utils.logger
-import config.Config
-import play.api.libs.json.{JsArray, JsObject, JsValue, Json}
-import play.api.mvc.{Action, Controller, Result}
+import com.kakao.s2graph.core.rest.RestCaller
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Controller, Request}
 
-import scala.concurrent._
 import scala.language.postfixOps
-import scala.util.Try
 
-object QueryController extends Controller {
+object QueryController extends Controller with JSONParser {
 
   import ApplicationController._
   import play.api.libs.concurrent.Execution.Implicits.defaultContext
 
-  private val s2: Graph = com.kakao.s2graph.rest.Global.s2graph
-  private val requestParser = com.kakao.s2graph.rest.Global.s2parser
+  private val rest: RestCaller = com.kakao.s2graph.rest.Global.s2rest
 
-  private def badQueryExceptionResults(ex: Exception) = Future.successful(BadRequest(Json.obj("message" -> ex.getMessage)).as(applicationJsonHeader))
+  def delegate(request: Request[JsValue]) =
+    rest.uriMatch(request.uri, request.body).map { js =>
```

Comment:
Is this figuring out which routes to call and actually call then wait for Future[JsValue]? then, uriMatch does not sounds good to me, since it actually make RPC call using Graph client. It is hard to guess that uriMatch will actually make call. What do you think?